src/bacnet/basic/object/iv.c

```diff
@@ -239,16 +241,46 @@ unsigned Integer_Value_Instance_To_Index(uint32_t object_instance)
 int32_t Integer_Value_Present_Value(uint32_t object_instance)
 {
     int32_t value = 0;
-    unsigned int index;
+    struct integer_object *pObject = Integer_Value_Object(object_instance);
 
-    index = Integer_Value_Instance_To_Index(object_instance);
-    if (index < MAX_INTEGER_VALUES) {
-        value = Integer_Value[index].Present_Value;
+    if (pObject) {
+        value = pObject->Present_Value;
     }
 
     return value;
 }
 
+/**
+ * This function is used to detect a value change,
+ * using the new value compared against the prior
+ * value, using a delta as threshold.
+ *
+ * This method will update the COV-changed attribute.
+ *
+ * @param index  Object index
+ * @param value  Given present value.
+ */
+static void
+Integer_Value_COV_Detect(struct integer_object *pObject, int32_t value)
+{
+    if (pObject) {
+        int32_t prior_value = pObject->Prior_Value;
+        uint32_t cov_increment = pObject->COV_Increment;
+        uint32_t cov_delta = 0;
+
+
+        if (prior_value > value) {
```

Comment:
since both values are signed int. This if check is unncessary as abs(value - prior_value) = abs(prior_value - value) is always trule.

Reply:
Re-worked...